specification/cognitiveservices/data-plane/Personalizer/stable/v1.0/Personalizer.json

```diff
@@ -0,0 +1,1258 @@
+      "type": "object",
+      "properties": {
+        "enableOfflineExperimentation": {
+          "description": "True if the evaluation should explore for a more optimal Learning settings.",
+          "type": "boolean"
+        },
+        "name": {
+          "description": "The name of the evaluation.",
+          "maxLength": 256,
+          "type": "string"
+        },
+        "startTime": {
+          "format": "date-time",
+          "description": "The start time of the evaluation.",
+          "type": "string"
+        },
+        "endTime": {
+          "format": "date-time",
+          "description": "The end time of the evaluation.",
+          "type": "string"
+        },
+        "policies": {
+          "description": "Additional Learning settings to evaluate.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PolicyContract"
+          }
+        }
+      }
+    },
+    "RewardRequest": {
+      "description": "Reward given to a rank response.",
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "format": "float",
+          "description": "Reward to be assigned to an action. Value should be between -1 and 1 inclusive.",
+          "type": "number"
+        }
+      }
+    },
+    "LogsProperties": {
+      "type": "object",
+      "properties": {
+        "dateRange": {
+          "allOf": [
+            {
+              "$ref": "#/definitions/DateRange"
+            }
+          ],
+          "readOnly": true
+        }
+      }
+    },
+    "DateRange": {
+      "type": "object",
+      "properties": {
+        "from": {
+          "format": "date-time",
+          "type": "string",
+          "readOnly": true
+        },
+        "to": {
+          "format": "date-time",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "ModelProperties": {
+      "type": "object",
+      "properties": {
+        "creationTime": {
+          "format": "date-time",
+          "type": "string",
+          "readOnly": true
+        },
+        "lastModifiedTime": {
+          "format": "date-time",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "RankRequest": {
+      "description": "Request a set of actions to be ranked by the Personalizer service.",
+      "required": [
+        "actions"
+      ],
+      "type": "object",
+      "properties": {
+        "contextFeatures": {
+          "description": "Features of the context used for Personalizer as a\r\ndictionary of dictionaries. This depends on the application, and\r\ntypically includes features about the current user, their\r\ndevice, profile information, aggregated data about time and date, etc.\r\nFeatures should not include personally identifiable information (PII),\r\nunique UserIDs, or precise timestamps.",
+          "type": "array",
+          "items": {
+            "type": "object"
```

Comment:
Would it be possible to have a map of objects rather than an array of objects? Or, even more constrained, have a map of map of strings? As it is, this API is extremely open ended w. regards to what I can provide.

Reply:
@sharathm@microsoft.com can you take a look at this ?

---
In reply to: [344972995](https://github.com/Azure/azure-rest-api-specs/pull/7624#discussion_r344972995) [](ancestors = 344972995)

Reply:
Wont fix for now. We are aware of this constraint. We have existing customers that use this api.

---
In reply to: [345552843](https://github.com/Azure/azure-rest-api-specs/pull/7624#discussion_r345552843) [](ancestors = 345552843,344972995)